tests/test_scripts.py

```diff
@@ -0,0 +1,22 @@
+import runpy
+
+import pytest
+from pathlib import Path
+
+
+ROOT_DIR = Path(__file__).parent.parent
+
+
+class TestExamples:
+    """
+    A class to test the example scripts.
+    """
+
+    def list_of_files():
+        file_list = (ROOT_DIR / "examples" / "scripts").rglob("*.py")
+        return file_list
+
+    @pytest.mark.parametrize("files", list_of_files())
+    @pytest.mark.scripts
+    def test_example_scripts(self, files):
+        runpy.run_path(files)
```

Comment:
One thing I'm finding difficult with this approach right now is about seeing the names of the files – `pytest` displays just numbers, and the `-s` option means that it prints out a lot of information, coupled with the additional logging we've set through the fixture. Is it possible to maintain a clear separation between the file names and actually print the file names in the test summary info at the end? You may refer to the difference between the current logs and the logs in other PRs. I don't think the verbose logging is helping all that much – scrolling through 14K+ lines of logs isn't useful to us.

Reply:
I agree with that, unit tests logs are also a little too much ~75,000.

About improving the clarity of test output, maybe we can try modifying the way pytest parametrize displays the file names.

Instead of just displaying numbers, we can try displaying a more descriptive name by using `pytest.param` while returning file list, try returning a `pytest.param` object with an id set to the file name like `[pytest.param(file, id=file.name) for file in file_list]`

Reply:
This should be resolved by https://github.com/pybamm-team/PyBaMM/pull/4180#discussion_r1693373168.